### PR TITLE
I have implemented the authentication functionality. The user connect…

### PR DESCRIPTION
**Pull request by AlexandraBulaceanu**
…s with the google account. He can then save the itinerary created in the Datastore. Also, a person can visualize all the previous trips that were created on that account and can remove trips. I have made changes in the design of the pages.

### CODE REVIEW

**Review by hasanbal — APPROVED** [review]

File: frontend/src/Map.js [hunk 1]
```diff
@@ -2,42 +2,50 @@ import React, { Component } from 'react';
...
-
             const pair = vars[i].split("=");
             const key = decodeURIComponent(pair[0]);
             const value = decodeURIComponent(pair[1]);
             query_string[key] = value;
         }
+
         let urlParameters = query_string;
-        const centerPoint = urlParameters.locations.split('*')[0];
+        //const centerPoint = urlParameters.locations.split('*')[0];
```
Comment (maffina) [inline frontend/src/Map.js:36]: please remove commented lines

File: frontend/src/Map.js [hunk 2]
```diff
@@ -62,15 +69,23 @@ class Map extends Component {
...
+            optimizeWaypoints: true,
+            travelMode: window.google.maps.TravelMode.DRIVING
+        };
 
+        this.m_get_directions_route(delayFactor, request, directionsRenderer, directionsService);
+
+        this.props.onMapLoad(map)
+    }
+
+    m_get_directions_route(delayFactor, request, directionsRenderer, directionsService) {
```
Comment (maffina) [inline frontend/src/Map.js:86]: tbh, it's unclear for me from the name, what this function does. Consider renaming to something more clear

**Review by maffina — APPROVED** [review]
Looks good in general, apart from linters and a couple of nits

File: frontend/src/Map.js (hunk 2 above)
Comment (AlexandraBulaceanu) [reply frontend/src/Map.js:86]: It generates the routes from the api, sure, I will change the name

File: frontend/src/Map.js (hunk 1 above)
Comment (AlexandraBulaceanu) [reply frontend/src/Map.js:36]: Yes, thank you

File: frontend/src/LandingPage.js [hunk 3]
```diff
@@ -6,19 +6,18 @@ import { Link } from 'react-router-dom';
...
 import Button from "react-bootstrap/Button";
 import ReactDOM from 'react-dom';
 import ImgKey from './configImg';
 import DayPickerInput from 'react-day-picker/DayPickerInput';
 import 'react-day-picker/lib/style.css';
 
-let autoComplete1, autoComplete2, currentUserId;
-let lat1, lat2, long1, long2, url;
+let autoComplete1, autoComplete2, autoComplete3, currentUserId;
+let lat1, lat2, long1, long2, lat3, long3, url;
```
Comment (maffina) [inline frontend/src/LandingPage.js:16]: The names that only differ in one letter, discouraged. It's easy to make a typo here. Consider renaming it to something more describing.

**Pull request merged**